traffic_ops/traffic_ops_golang/iso/iso.go

```diff
@@ -0,0 +1,368 @@
+
+	body, err := json.Marshal(alerts)
+	if err != nil {
+		statusCode := http.StatusInternalServerError
+		api.HandleErr(w, req, nil, statusCode, errors.New(http.StatusText(statusCode)), fmt.Errorf("error marshalling JSON: %v", err))
+		return
+	}
+
+	w.Header().Set(rfc.ContentType, rfc.ApplicationJSON)
+	w.WriteHeader(http.StatusBadRequest)
+	w.Write(body)
+}
+
+// isoRequest represents the JSON object clients use to
+// request an ISO be generated.
+type isoRequest struct {
+	OSVersionDir  string          `json:"osversionDir"`
+	HostName      string          `json:"hostName"`
+	DomainName    string          `json:"domainName"`
+	RootPass      string          `json:"rootPass"`
+	DHCP          boolStr         `json:"dhcp"`
+	IPAddr        net.IP          `json:"ipAddress"`
+	IPNetmask     net.IP          `json:"ipNetmask"`
+	IPGateway     net.IP          `json:"ipGateway"`
+	IP6Address    net.IP          `json:"ip6Address"`
+	IP6Gateway    net.IP          `json:"ip6Gateway"`
+	InterfaceName string          `json:"interfaceName"`
+	InterfaceMTU  util.JSONIntStr `json:"interfaceMtu"`
+	Disk          string          `json:"disk"`
+	MgmtIPAddress net.IP          `json:"mgmtIpAddress"`
+	MgmtIPNetmask net.IP          `json:"mgmtIpNetmask"`
+	MgmtIPGateway net.IP          `json:"mgmtIpGateway"`
+	MgmtInterface string          `json:"mgmtInterface"`
+	Stream        boolStr         `json:"stream"`
+}
+
+func (i *isoRequest) fqdn() string {
+	fqdn := i.HostName
+	if i.DomainName != "" {
+		fqdn += "." + i.DomainName
+	}
+	return fqdn
+}
+
+// validate returns an empty slice if the isoRequest is valid. Otherwise,
+// it returns a slice of error messages.
+func (i *isoRequest) validate() []string {
+	var errs []string
+	addErr := func(msg string) { errs = append(errs, msg) }
+
+	if i.OSVersionDir == "" {
+		addErr("osversionDir is required")
+	}
+	if i.HostName == "" {
+		addErr("hostName is required")
+	}
+	if i.DomainName == "" {
+		addErr("domainName is required")
+	}
+	if i.RootPass == "" {
+		addErr("rootPass is required")
+	}
+	if !i.DHCP.isSet {
+		addErr("dhcp is required")
+	}
+	if i.InterfaceMTU == 0 {
+		addErr("interfaceMtu is required")
+	}
+	if i.Disk == "" {
+		addErr("disk is required")
+	}
+	if len(i.MgmtIPAddress) > 0 {
+		if i.MgmtInterface == "" {
+			addErr("mgmtInterface is required when mgmtIpAddress is provided")
+		}
+		if len(i.MgmtIPGateway) == 0 {
+			addErr("mgmtIpGateway is required when mgmtIpAddress is provided")
+		}
+	}
+	if v, ok := i.DHCP.val(); ok && !v {
+		if len(i.IPAddr) == 0 {
+			addErr("ipAddress is required if DHCP is no")
+		}
+		if len(i.IPNetmask) == 0 {
+			addErr("ipNetmask is required if DHCP is no")
+		}
+		if len(i.IPGateway) == 0 {
+			addErr("ipGateway is required if DHCP is no")
+		}
+	}
+
+	// Use of stream is deprecated it will always stream
+	// regardless if this is set or not.
+	if v, ok := i.Stream.val(); ok && !v {
+		addErr("stream request parameter is deprecated")
+	}
```

Comment:
No reason to check/allow/parse this, because the handler only services API 2.x

Reply:
my thought was to catch ppl trying to use it in a legacy manner but our other endpoints do that so yeah removing